src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -502,10 +505,18 @@ private[kafka] object KafkaConsumerActor {
 
     def withFetch(
       partition: TopicPartition,
+      streamId: Int,
       deferred: Deferred[F, (Chunk[CommittableMessage[F, K, V]], FetchCompletedReason)]
     ): State[F, K, V] = {
-      val fetch = NonEmptyChain.one(FetchRequest(deferred))
-      copy(fetches = fetches combine Map(partition -> fetch))
+      val fetch = Map(streamId -> FetchRequest(deferred))
+
+      // revoke any previous fetch on the same (partition,streamId)
+      // TODO log a warning as this shouldn't happen!
+      val existing = fetches.get(partition).flatMap(_.get(streamId))
+      existing.foreach(_.completeRevoked(Chunk.empty))
+
+      val fetchesForPartition = fetches.getOrElse(partition, Map.empty) ++ fetch
```

Comment:
We've already done the `Map` lookup as part of `existing`, so we could reuse it here. We could also use `updated` to avoid creating the extra `Map` for `fetch`.

Reply:
Sure